app/models/automated_checks/claim_verifiers/identity.rb

```diff
@@ -14,9 +14,10 @@ def initialize(
       def perform
         return unless awaiting_task?("identity_confirmation")
 
-        complete_match ||
-          partial_match ||
-          no_match
+        # Order matters and seems to create unnecessary conditionals
```

Comment:
@nomasprime could you maybe elaborate on this somewhere? So when we come back to look at code in future we have some background? It feels like this might be some kind of tech debt?

Reply:
Not sure how to elaborate because I don't know without actually doing the refactor? It's just the ORs and subsequent conditionals aren't very manageable.

Suggested when this work was started it followed a similar design to Rails validations, not sure why it didn't take that direction but imagine it would end up something like that.